python/kserve/kserve/storage/storage.py

```diff
@@ -211,19 +211,25 @@ def _download_s3(uri, temp_dir: str) -> str:
             verify_ssl = True
 
         # If verify_ssl is true, then check there is custom ca bundle cert
+        # The CA bundle can be any local file in the container under the path
+        # set in the AWS_CA_BUNDLE environment variable.
+        # It can also be coming from a ConfigMap, in which case, the filename
+        # cabundle.crt.
         if verify_ssl:
             global_ca_bundle_configmap = os.getenv("CA_BUNDLE_CONFIGMAP_NAME")
-            if global_ca_bundle_configmap:
-                isvc_aws_ca_bundle_path = os.getenv("AWS_CA_BUNDLE")
-                if isvc_aws_ca_bundle_path and isvc_aws_ca_bundle_path != "":
-                    ca_bundle_full_path = isvc_aws_ca_bundle_path
-                else:
-                    global_ca_bundle_volume_mount_path = os.getenv(
-                        "CA_BUNDLE_VOLUME_MOUNT_POINT"
-                    )
-                    ca_bundle_full_path = (
-                        global_ca_bundle_volume_mount_path + "/cabundle.crt"
-                    )
+            isvc_aws_ca_bundle_path = os.getenv("AWS_CA_BUNDLE")
```

Comment:
The AWS_CA_BUNDLE environment variable is used to specify the path to the CA file, under the assumption that the CA is already present inside the pod. However, shouldn’t there be a logic to check whether this file actually exists?

Also, in the OpenShift use case, it's said that the service-ca.crt file is always present, but in reality, this is controlled by configuration. For example, in this [link](https://github.com/kserve/kserve/blob/master/config/configmap/inferenceservice.yaml#L721), if the value is set to false, the file is not attached to the pod. Our team discussed whether we really can remove this file from runtime. @VedantMahabaleshwarkar can you please share the result ?

Reply:
There is a check for the file existence lower. I am just splitting the logic between `AWS_CA_BUNDLE` and `CA_BUNDLE_VOLUME_MOUNT_POINT` environment variables. Then, the `if os.path.exists(ca_bundle_full_path):` block is checking the file.

Reply:
Yes I missed the part! then I am ok with this change :)